### PR TITLE
RavenDB-9479 - support for control code characters

### PR DESCRIPTION
**Pull request by grisha-kotler**
http://issues.hibernatingrhinos.com/issue/RavenDB-9479

### CODE REVIEW

File: test/FastTests/Issues/RavenDB-9479.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                        Letter = ch
+                    };
+
+                    await session.StoreAsync(question);
+                    await session.SaveChangesAsync();
+
+                    documentId = question.Id;
+                }
+
+                WaitForUserToContinueTheTest(store);
```
Comment (ml054) [inline test/FastTests/Issues/RavenDB-9479.cs:34]: Should we commit with this?

File: src/Sparrow/Json/LazyStringValue.cs [hunk 2]
```diff
@@ -903,5 +903,15 @@ public void Renew(string str, byte* buffer, int size)
...
         }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public bool IsControlCodeCharacter(out byte b)
+        {
+            Debug.Assert(Size == 1);
+
+            b = Buffer[0];
+            // control code characters
+            return b < 32 || (b >= 127 && b <= 159);
```
Comment (ayende) [inline src/Sparrow/Json/LazyStringValue.cs:914]: Why is a single line string special in this way? What happens if we have a space and then a control char?

File: test/FastTests/Issues/RavenDB-9479.cs (hunk 1 above)
Comment (ayende) [reply test/FastTests/Issues/RavenDB-9479.cs:34]: Ideally, not. But if a debugger is not attached this is a noop

File: src/Sparrow/Json/LazyStringValue.cs (hunk 2 above)
Comment (talweiss1982) [reply src/Sparrow/Json/LazyStringValue.cs:914]: @ayende Starting to look into this it seems that this is not the right place to handle this.
We should construct the LazyStringValue with those control character already escaped meaning we will need to go over all the places that construct the escaped character and make sure we escape control characters, right?

Comment (ayende) [reply src/Sparrow/Json/LazyStringValue.cs:914]: That should be the case when we build LZV, yes.

Comment (maximburyak) [reply src/Sparrow/Json/LazyStringValue.cs:914]: I think that we should not treat that on server side.
We did not validate incoming jsons and that train has already left the platform, due to backward compatability issues (imagine refusing requests from older clients by newer servers etc.)
We do need to make sure that the clients will send a valid json from now on

**Pull request merged**